types/src/transaction/mod.rs

```diff
@@ -1345,6 +1345,25 @@ impl TransactionToCommit {
         &self.transaction_info
     }
 
+    pub fn set_state_checkpoint_hash(&mut self, state_checkpoint_hash: HashValue) -> Result<()> {
+        let transaction_info = self.transaction_info();
+        ensure!(transaction_info.state_checkpoint_hash().is_none(),
+                "Can only set state_checkpoint_hash when it is None. transaction_info: {transaction_info}.");
+
+        let transaction_info_with_state_checkpoint_hash = TransactionInfo::new(
+            transaction_info.transaction_hash(),
+            transaction_info.state_change_hash(),
+            transaction_info.event_root_hash(),
+            Some(state_checkpoint_hash),
+            transaction_info.gas_used(),
+            transaction_info.status().clone(),
+        );
+
+        self.transaction_info = transaction_info_with_state_checkpoint_hash;
```

Comment:
why not update in place?

Reply:
Want to avoid exposing a mutable API directly from TransactionInfo, to limit the place to mutate it.

Reply:
Then a `TransactionInfo::fill_state_checkpoint_hash(mut self, hash: HashValue) -> Self` would be neat (then you can https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=aaba6e74c2baed5286d6a3431fd275b5). up to you.

Reply:
Just realized I need to update the hash before calculating the accumulator.. Now this function is not needed.